tests/ui/kindck/kindck-impl-type-params.stderr

```diff
@@ -112,13 +112,13 @@ LL |     struct Foo; // does not impl Copy
    |
 
 error: lifetime may not live long enough
-  --> $DIR/kindck-impl-type-params.rs:30:13
+  --> $DIR/kindck-impl-type-params.rs:30:19
    |
 LL | fn foo<'a>() {
    |        -- lifetime `'a` defined here
 LL |     let t: S<&'a isize> = S(marker::PhantomData);
 LL |     let a = &t as &dyn Gettable<&'a isize>;
-   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ type annotation requires that `'a` must outlive `'static`
+   |                   ^^^^^^^^^^^^^^^^^^^^^^^^ type annotation requires that `'a` must outlive `'static`
```

Comment:
question: why does this say "type annotation" instead of "cast" like most others?

Reply:
Because borrow errors from type annotations are given special priority over errors from other sources (including casts) here: https://github.com/rust-lang/rust/blob/1a5a2240bc1b8cf0bcce7acb946c78d6493a4fd3/compiler/rustc_borrowck/src/region_infer/mod.rs#L2105
Changing this code at all makes many other diagnostics worse, so I'd prefer to leave it alone for now.

Reply:
Also, this special casing can cause some diagnostic output that is just plainly wrong, so it should definitely be changed in the future.

```rust
trait Trait<'a> {}

fn require_static(_: *const dyn Trait<'static>) {}

fn not_static<'a>(a: *const dyn Trait<'a>) {
    require_static(a as *const dyn Trait<'a>);
}
```

```
error: lifetime may not live long enough
 --> src/lib.rs:6:20
  |
5 | fn not_static<'a>(a: *const dyn Trait<'a>) {
  |               -- lifetime `'a` defined here
6 |     require_static(a as *const dyn Trait<'a>);
  |                    ^^^^^^^^^^^^^^^^^^^^^^^^^ type annotation requires that `'a` must outlive `'static`
```